### ISSUE TITLE
Started Once container returning non-zero is not reporting errors

### ISSUE THREAD

**Issue by stevelowery** [rancher/rancher#5978]
**Rancher Version:** 1.1.3

**Docker Version:** 1.11.2

**OS and where are the hosts located? (cloud, bare metal, etc):** CentOS Linux 7, Bare metal

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)** Single Node

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)** Swarm

**Steps to Reproduce:**  Start a container with a label io.rancher.container.start_once: 'true' that exits with a non-zero code.

**Results:** When doing a docker ps -a and viewing the container, you can see that the container exited with the given code.  Rancher does not attempt to restart it (we told it not to), but it doesn't show that the container exited with an error status.  The only way to know something bad happened is to go into the docker logs.

**Expected:**  I would expect that rancher would report the container in a state of Error or Unhealthy or something to the like to indicate there was a problem when that container ran.

**Comment by nagliyvred** [comment]
Is there a workaround for this to determine the exit code until this bug is fixed?

**Comment by stevelowery** [comment]
If there is a workaround, I am not aware of it.  From the Rancher apis, it you cannot tell the difference for a started once container that exited with a non zero exit code and one that exists succesfully.  All we can really do is view the logs and see if there are any errors generated during the run.  Makes automating deployment/upgrade of these types of services impossible.

**Comment by alena1108** [comment]
@cjellick would it be possible to propagate a complete "state" field from docker inspect, to Rancher? So in the API we see this kind of information:

```
 "State": {
            "Status": "exited",
            "Running": false,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "Status": "exited",
            "Running": false,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2016-12-15T22:12:53.548083973Z",
            "FinishedAt": "2016-12-15T22:12:53.629192874Z"
        },
```

**Comment by NathanRSmith** [comment]
+1

**Comment by mrproper** [comment]
+1

We use start_once to run db seeds and migrations, if they fail to run for some reason, theres no visibility that it has not happened short of checking the logs for the service.

What i would expect is the start_once is only "start-once" state if the service ran with a 0 exit code otherwise its degraded

**Comment by cjellick** [comment]
bumping out of september. After reviewing, we realized a flaw in the design of the fix. Need to revisit later.

**Comment by mcanevet** [comment]
@deniseschannon Is there any hope to see this bug fixed in v1.6 now that v2.0 is out?

**Comment by deniseschannon** [comment]
With the release of Rancher 2.0, development on v1.6 is only limited to critical bug fixes and security patches.
